### PR TITLE
logger: add_topic_multi optional specify max multi-instance

### PR DESCRIPTION
**Pull request by dagar**
Looking for ways to be able to affordably increase `ORB_MULTI_MAX_INSTANCES`.

### CODE REVIEW

**Comment by dagar** [comment]
The other option I considered was at topic initialization time to add the first instance of a multi-topic, then only additional instances if they're advertised. Obviously not ideal, but it's cheap from a logger perspective and good enough for most cases.

File: src/modules/logger/logged_topics.h [hunk 1]
```diff
@@ -110,7 +110,16 @@ class LoggedTopics
...
 	 */
 	bool add_topic(const char *name, uint16_t interval_ms = 0, uint8_t instance = 0);
-	bool add_topic_multi(const char *name, uint16_t interval_ms = 0);
+
+	/**
+	 * Add a topic to be logged.
+	 * @param name topic name
+	 * @param interval limit in milliseconds if >0, otherwise log as fast as the topic is updated.
+	 * @param instance orb topic instance
+	 * @param max_instance the max multi-instance to add.
```
Comment (bkueng) [inline src/modules/logger/logged_topics.h:119]: This is fairly nitpicky, but I would understand this as the maximum instance index to add (`instance <= max_instance`), not the max count. `max_num_instances` would IMO be clearer.

**Review by bkueng — COMMENTED** [review]
This is good.
Have you looked into adding the max count to the topic definition and use that at runtime? I assume this would not be much effort either, but maybe I'm missing something.

**Comment by dagar** [comment]
> Have you looked into adding the max count to the topic definition and use that at runtime? I assume this would not be much effort either, but maybe I'm missing something.

Yes, I tried adding a special comment field to optionally set the max number of instances. That worked okay other than some multi-topic edge cases where I wanted different limits. I stopped because the entire idea of the topics and a growing list of metadata coming from the msg files directly was really starting to feel overloaded. I then started thinking about doing it explicitly as a table somewhere else.

 - topic name, type, max instances (optional), queue size (optional)

Overlaps with some of the discussion in https://github.com/PX4/Firmware/pull/15610.

File: src/modules/logger/logged_topics.h (hunk 1 above)
Comment (dagar) [reply src/modules/logger/logged_topics.h:119]: Please nitpick away, I wasn't sure about the naming here.

Comment (dagar) [reply src/modules/logger/logged_topics.h]: Renamed.

**Review by bkueng — APPROVED** [review]

**Pull request merged**